Updates API examples and tests

m1kola opened the pull request:
* Makes tests to not rely on examples. Currently changes in examples can cause tests to fail. Hopefully without dependency on examples there will be less false positives.
* Also changes examples structure: `pkg/api` now has an examole of `OpenShiftClusterDocument` with the `OpenShiftCluster` field populated. Specific API versions can now convert this (internal) model into their representation.

jim-minter commented on pkg/api/openshiftclusterdocument_example.go:
```diff
@@ -0,0 +1,78 @@
+		Key:                       "/subscriptions/subscriptionId/resourceGroups/resourceGroup/providers/Microsoft.RedHatOpenShift/OpenShiftClusters/resourceName",
+		Bucket:                    42,
+		ClusterResourceGroupIDKey: "/subscriptions/subscriptionId/resourceGroups/clusterResourceGroup",
+		ClientIDKey:               "11111111-1111-1111-1111-111111111111",
```
note that these key fields must very specifically be lower case.

jim-minter commented on pkg/api/v20191231preview/openshiftcluster_example.go:
```diff
@@ -3,60 +3,18 @@ package v20191231preview
+	oc, ok := (&openShiftClusterConverter{}).ToExternal(doc.OpenShiftCluster).(*OpenShiftCluster)
+	if !ok {
+		panic("Can not convert types")
 	}
```
without the `, ok` above you will also get a panic - the `, ok` doesn't really add anything

jim-minter commented:
> Hopefully without dependency on examples there will be less false positives.

Could you give a bit more detail?  I'm prepared to agree, but I'm not yet caught on to why duplicating is better here?

jim-minter commented:
@m1kola bump on question above

m1kola commented:
@jim-minter I addressed the feedback and rebased on top of the master.

> Could you give a bit more detail? I'm prepared to agree, but I'm not yet caught on to why duplicating is better here?

So when I was working on other PRs (mostly admin API auth) I had few test failures not due to changes to the code, but to the changes in the examples. I think the more we rely on examples in tests the more issues like this we will be noticing.
